src/HttpLayer.hs

```diff
@@ -40,41 +42,62 @@ emptyHTTPNetworkLayer = HTTPNetworkLayer
 ------------------------------------------------------------
 
 -- | General api request function
-apiRequest :: Config -> Text -> Request
-apiRequest Config{..} u =
-    setRequestPath (encodeUtf8 path) $
-    addRequestHeader "Content-Type" "application/json" $
-    setRequestBasicAuth
-        (encodeUtf8 cfgEmail <> "/token")
-        (encodeUtf8 cfgToken) $
-    parseRequest_ (toString (cfgZendesk <> path))
+apiRequest
+    :: Config
+    -> Text
+    -> Either String Request
+apiRequest Config{..} u = mapLeft show $ catches buildRequest handlerList
   where
+    buildRequest :: forall m. MonadCatch m => m Request
+    buildRequest = do
+        req <- parseUrlThrow (toString (cfgZendesk <> path)) -- TODO(md): Get a list of exceptions that parseUrlThrow can throw
+        return $ setRequestPath (encodeUtf8 path) $
+                 addRequestHeader "Content-Type" "application/json" $
+                 setRequestBasicAuth
+                     (encodeUtf8 cfgEmail <> "/token")
+                     (encodeUtf8 cfgToken) req
+    handlerList :: MonadCatch m => [Handler m Request]
+    handlerList = [
+                    handlerJSON
+                  , handlerHTTP
+                  ]
+    handlerJSON :: MonadCatch m => Handler m Request
+    handlerJSON = Handler $ \(ex :: JSONException) -> error $ show ex
+    handlerHTTP :: MonadCatch m => Handler m Request
+    handlerHTTP = Handler $ \(ex :: HttpException) -> error $ show ex
     path :: Text
     path = "/api/v2" <> u
 
 -- | Api request but use absolute path
-apiRequestAbsolute :: Config -> Text -> Request
+apiRequestAbsolute
+    :: Config
+    -> Text
+    -> Either String Request
```

Comment:
Same here.

Reply:
Same here, as in "think why you return the `Either`". Why do you return it?

Reply:
I updated the implementation of this function. Now it is almost the same as the `apiRequest` function, hence it is clear why `Either` in the return type.